Common/src/Common/Controller/Lva/AbstractController.php

```diff
@@ -381,17 +382,16 @@ protected function getLicenceId($applicationId = null)
      *
      * @return bool
      */
-    protected function isExternal()
+    protected function isExternal(): bool
     {
         return $this->location === self::LOC_EXTERNAL;
     }
 
     /**
-     * Return true if the current internal user has read only permissions
-     *
-     * @return bool
+     * Have left this in place for now as the number of controllers extending it made removal impractical
+     * @see Permission for the preferred way to access this logic via dependency injection
```

Comment:
Should we add `@deprecated` annotation and point to the new `Permission` service here?

Reply:
Yes - added the annotation. The comment did point to the permission service (in the sense the IDE could follow it) but have swapped to a FQCN so it's more readable to human eye.